### PR TITLE
create-cluster-kubeadm: update the version skew policy

### LINKED ISSUE #32304: kubeadm: improve the version skew documentation
in this section:
https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/#version-skew-policy

we document the kubeadm skew against the control plane, but we do not mention the skew of kubeadm against:
- the kubelet
- kubeadm itself

replace the section with the following to clarify all points:

```
## Version skew policy {#version-skew-policy}

While kubeadm allows version skew against some components that it manages, it is recommended that you
match the kubeadm version with the versions of the control plane components, kube-proxy and kubelet.

### kubeadm's skew against the Kubernetes version

kubeadm can be used with Kubernetes components that are the same version as kubeadm
or one version older. The Kubernetes version can be specified to kubeadm by using the
`--kubernetes-version` flag of `kubeadm init` or the
[`ClusterConfiguration.kubernetesVersion`](/docs/reference/config-api/kubeadm-config.v1beta3/)
field when using `--config`. This option will control the versions
of kube-apiserver, kube-controller-manager, kube-scheduler and kube-proxy.

Example:
* kubeadm is at {{< skew latestVersion >}}
* `kubernetesVersion` must be at {{< skew latestVersion >}} or {{< skew prevMinorVersion >}}

### kubeadm's skew against the kubelet

Similarly to the Kubernetes version, kubeadm can be used with a kubelet version that is the same
version as kubeadm or one version older.

Example:
* kubeadm is at {{< skew latestVersion >}}
* kubelet on the host must be at {{< skew latestVersion >}} or {{< skew prevMinorVersion >}}

### kubeadm's skew against kubeadm

There are certain limitations on how kubeadm commands can operate on existing nodes or whole cluster
managed by kubeadm.

If new nodes are joined to the cluster, the kubeadm binary used for `kubeadm join` must match
the last version of kubeadm used to either create the cluster with `kubeadm init` or to upgrade
the same node with `kubeadm upgrade`. Similar rules apply to the rest of the kubeadm commands
with the exception of `kubeadm upgrade`.

Example for `kubeadm join`:
* kubeadm version {{< skew latestVersion >}} was used to create a cluster with `kubeadm init`
* Joining nodes must use a kubeadm binary that is at version {{< skew latestVersion >}}

Nodes that are being upgraded must use a version of kubeadm that is the same MINOR
version or one MINOR version newer than the version of kubeadm used for managing the
node.

Example for `kubeadm upgrade`:
* kubeadm version {{< skew prevMinorVersion >}} was used to create or upgrade the node
* The version of kubeadm used for upgrading the node must be at {{< skew prevMinorVersion >}}
or {{< skew latestVersion >}}

To learn more about the version skew between the different Kubernetes component see
the [Version Skew Policy](https://kubernetes.io/releases/version-skew-policy/).
```

### PR DESCRIPTION
**Pull request by PriyanshuAhlawat**
Version skew policy has been updated and now it also mentions kubeadm's skew against
- the kubelet
- kubeadm itself
closes #32304

### CODE REVIEW

**Comment by neolit123** [comment]
i guess self-LGTM, given i wrote the text. thanks @PriyanshuAhlawat for sending the PR.

this PR updates the kubeadm skew policy to reflect the currently supported state.
comments about the clarity of this policy are appreciated!
something that is missing is potential note about immutable upgrades. CAPI and some users do immutable node-replace upgrade, which forces the actor to implement portions of "kubeadm upgrade apply" or otherwise kubeadm N+1 might not be able to join to a N cluster.

PTAL for technical review:
/cc @dlipovetsky @sbueringer @fabriziopandini 

for sig-docs approval:
/assign @sftim

**Comment by neolit123** [comment]
related slack discussion with @dlipovetsky 
https://kubernetes.slack.com/archives/C2P1JHS2E/p1647374486580179

/retitle create-cluster-kubeadm: update the version skew policy

File: content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md [hunk 1]
```diff
@@ -524,15 +524,56 @@ options.
...
+Similarly to the Kubernetes version, kubeadm can be used with a kubelet version that is the same
+version as kubeadm or one version older.
+
+Example:
+* kubeadm is at {{< skew latestVersion >}}
+* kubelet on the host must be at {{< skew latestVersion >}} or {{< skew prevMinorVersion >}}
+
+### kubeadm's skew against kubeadm
+
+There are certain limitations on how kubeadm commands can operate on existing nodes or whole cluster
```
Comment (sbueringer) [inline content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md:554]:
```suggestion
There are certain limitations on how kubeadm commands can operate on existing nodes or whole clusters
```
nit, I think

**Review by sbueringer — COMMENTED** [review]
Sorry for the late review

Very clear documentation, thank you!

/lgtm

**Comment by neolit123** [comment]
> I think it's fine to merge as is as it makes the support policy of kubeadm itself very clear. We might want to follow-up with some additional notes about consumers like CAPI, but I think in CAPI the situation itself is absolutely non-trivial.

Agreed. We can follow on notes in separate PRs.

/assign @reylejano
PTAL for approval.

/lgtm cancel
I can reapply the lgtm once the typo is fixed.

**Pull request merged**